pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,290 @@
+        self._host = split_address[0]
+        self._port = int(split_address[1])
+
+        # Max file descriptors allowed by the OS
+        self._max_connections = max_connections if max_connections else os.sysconf("SC_OPEN_MAX")
+        self.selector = selectors.DefaultSelector()
+        self.connections = {}  # Track active sockets
+        self.tasks = {}  # Track active asyncio tasks
+        self.shutdown_event = asyncio.Event()
+
+    async def _handle_event(self, event, sock):
+        # take call time
+        start_time = time.time()
+
+        # call the entrypoint
+        entrypoint_output = await self._entrypoint(self._context, event)
+
+        # measure duration, set to minimum float in case execution was too fast
+        duration = time.time() - start_time or sys.float_info.min
+
+        await self._write_packet_to_processor(sock, 'm' + json.dumps({'duration': duration}))
+
+        # try to json encode the response
+        encoded_response = self._encode_entrypoint_output(entrypoint_output)
+
+        # write response to the socket
+        await self._write_packet_to_processor(sock, 'r' + encoded_response)
+
+    async def _process_connection(self, sock):
+        """Process all events for a single connection."""
+        # signal start
+        self._logger.info("Signalling connection processing start")
+        await self._write_packet_to_processor(sock, 's')
+        self._logger.info(f"Event processing started for socket")
+        try:
+            while True:
+                # resolve event message length
+                event_message_length = await self._resolve_event_message_length(sock)
+                # Resolve event message
+                event = await self._resolve_event(sock, event_message_length)
+
+                # Handle the event
+                if not self._discard_events:
+                    try:
+                        await self._handle_event(event, sock)
+                    except BaseException as exc:
+                        await self._on_handle_event_error(exc, sock)
+                else:
+                    self._logger.debug("Event discarded")
+
+                # Release event reference
+                del event
+
+        except (ConnectionResetError, asyncio.IncompleteReadError):
+            self._logger.info("Client disconnected")
+        except WrapperFatalException as exc:
+            await self._on_serving_error(exc, sock)
+
+            # explode, unrecoverable exception
+            await self._shutdown(error_code=1)
+
+        except UnicodeDecodeError as exc:
+
+            # reset unpacker to avoid consecutive errors
+            # this may happen when msgpack fails to decode a non-utf8 events
+            self._unpacker = self._resolve_unpacker()
+            await self._on_serving_error(exc, sock)
+
+        except asyncio.CancelledError:
+            self._logger.debug('Waiting for event message was interrupted by a signal')
```

Comment:
Not sure the cancellation is only for "Waiting for an event" like it is in the regular wrapper, as now you can also cancel the entire `_process_connection` task.